### PR TITLE
Cli2restart -- See if this is close

### PR DESCRIPTION
**Pull request by GerryG**
Was loading and running tests, but my local DB was messing up.

### CODE REVIEW

**Comment by GerryG** [comment]
Oh, maybe it is the updates I didn't merge yet.

**Comment by GerryG** [comment]
NVM, got it to fail ... it isn't loading right for shark decks.

File: card/lib/card/config/environments/cucumber.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# -*- encoding : utf-8 -*-
+
+Cardio::Application.configure do
+end
```
[the remark this answers is not in the document]
Comment (ethn) [reply card/lib/card/config/environments/cucumber.rb:4]: If something is breaking when these aren't here, that's a bug.

File: card/lib/card/env/location.rb [hunk 2]
```diff
@@ -12,7 +12,7 @@ def card_path rel_path
         if rel_path =~ %r{^(https?\:)?/}
           rel_path
         else
-          "#{Card.config.relative_url_root}/#{rel_path}"
```
Comment (ethn) [inline card/lib/card/env/location.rb:15]: if you've already loaded Card, then Card.config is still an acceptable alternative to Cardio.config, right?  And if you're in a subclass of Card, you've already loaded Card, right?

File: card/lib/card/essential.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+
+require 'card/mark'
+require 'card/dirty/method_factory'
+require 'card/dirty'
+require 'card/director'
```
Comment (ethn) [inline card/lib/card/essential.rb:5]: this is also bad. please remove.

File: card/lib/card/name/fields_and_traits.rb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 class Card
-  class Name
+  class Name < Cardname
```
Comment (ethn) [inline card/lib/card/name/fields_and_traits.rb:2]: not necessary, please remove.

File: card/lib/cardio.rb [hunk 5]
```diff
@@ -19,9 +19,32 @@ module Cardio
   extend Delaying
   CARD_GEM_ROOT = File.expand_path("..", __dir__)
 
-  mattr_reader :paths, :config
+  module RailsConfigMethods
```
Comment (ethn) [inline card/lib/cardio.rb:22]: modules like this should be in cardio subdir.  (eg cardio/rails_config_methods.rb). Those will need to be explicitly required.

File: card/lib/cardio/commands.rb [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+require "rake"
+def load_rake_tasks
+  require "./config/environment"
+  require "rake"
+  Cardio::Application.load_tasks
+end
+
+module Cardio
+  module Commands
```
Comment (ethn) [inline card/lib/cardio/commands.rb:13]: These changes leave a TON of duplicate code in decko and card.  Anything that's already in cardio should be includable in decko.

File: card/spec/spec_helper.rb [hunk 7]
```diff
@@ -4,6 +4,7 @@
 CardSpecLoader.init
 
 require "rr"
+require "decko/commands/application"
```
Comment (ethn) [inline card/spec/spec_helper.rb:7]: should not be necessary here.

File: card/lib/cardio/application.rb [hunk 8]
```diff
@@ -0,0 +1,44 @@
+# -*- encoding : utf-8 -*-
+
+require 'rails'
```
Comment (ethn) [inline card/lib/cardio/application.rb:3]: too general. Even decko doesn't require "rails"

File: card/lib/cardio/application_record.rb [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+ActiveSupport.run_load_hooks(:before_application_record, self)
+
+module Cardio
+  class ApplicationRecord < ActiveRecord::Base
+    self.abstract_class = true
+  end
+end
+
+ActiveSupport.run_load_hooks(:after_application_record, self)
```
Comment (ethn) [inline card/lib/cardio/application_record.rb:11]: I think I would just leave this stuff out for now.  out of scope.

File: card/lib/cardio/commands/rake_command/parser.rb [hunk 10]
```diff
@@ -1,14 +1,14 @@
...
+module Cardio
   module Commands
     class RakeCommand
       class Parser < OptionParser
         def initialize command, opts
           super() do |parser|
-            parser.banner = "Usage: decko #{command} [options]\n\n" \
-                          "Run decko:#{command} task on the production "\
+            parser.banner = "Usage: card #{command} [options]\n\n" \
+                          "Run card:#{command} task on the production "\
```
Comment (ethn) [inline card/lib/cardio/commands/rake_command/parser.rb:11]: does this mean monkeys will be seeing help text for the `card` command? We may need to override this in decko.

File: card/lib/cardio/config/initializers/secret_token.rb [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+# no regular words or you'll be exposed to dictionary attacks.
+
+Rails.application.configure do
+  config.secret_token = "65632e048d6c80d1e63c911e1a40a51072413543f3182e0261"\
+                       "b52e3812b2c9f0ee81828fa5688a34c13a78e438a4b56f4971"\
+                       "35dc079a4e4460733d555968a2f8"
+  config.secret_key_base = "65632e048d6c80d1e63c911e1a40a51072413543f3182e0"\
+                           "261b52e3812b2c9f0ee81828fa5688a34c13a78e438a4b5"\
+                           "6f497135dc079a4e4460733d555968a2f8"
+end
```
Comment (ethn) [inline card/lib/cardio/config/initializers/secret_token.rb:16]: can't have this in repo.

File: card/lib/cardio/mod_inflector.rb [hunk 12]
```diff
@@ -0,0 +1,16 @@
+class ModInflector < Zeitwerk::Inflector
```
Comment (ethn) [inline card/lib/cardio/mod_inflector.rb:1]: shouldn't this be a `git mv` of old file?

File: decko/lib/decko/config/environments/cucumber.rb [hunk 13]
```diff
@@ -1,5 +1,5 @@
 # -*- encoding : utf-8 -*-
-Decko.application.class.configure do
+Decko::Application.configure do
```
Comment (ethn) [inline decko/lib/decko/config/environments/cucumber.rb:2]: This is wrong. 

When you make a deck, eg wikirate, it generates a new application class, eg WikirateApplication, which inherits from Decko::Application.  Decko.application.class returns WikirateApplication.

File: decko/lib/decko/config/environments/test.rb [hunk 14]
```diff
@@ -1,5 +1,5 @@
 # -*- encoding : utf-8 -*-
-Decko.application.class.configure do
+Decko::Application.configure do
```
Comment (ethn) [inline decko/lib/decko/config/environments/test.rb:2]: This might be what's breaking your specs.

**Review by ethn — CHANGES_REQUESTED** [review]

File: card-mod-defaults/card-mod-defaults.gemspec [hunk 15]
```diff
@@ -3,9 +3,9 @@
 require "../decko_gem"
 
 DeckoGem.mod "defaults" do |s, d|
-  d.depends_on_mod :ace_editor, :prosemirror_editor, :recaptcha, :tinymce_editor, :follow,
-                   :markdown, :date, :google_analytics, :carrierwave, :bootstrap,
-                   :account, :history, :delayed_job
+  d.depends_on_mod :ace_editor, :prosemirror_editor, :recaptcha,
+    :tinymce_editor, :follow, :markdown, :date, :google_analytics,
+    :carrierwave, :bootstrap, :account, :history, :delayed_job
```
Comment (ethn) [inline card-mod-defaults/card-mod-defaults.gemspec:8]: old version is correct, according to our current rubocop config.

File: card/lib/card/env/location.rb (hunk 2 above)
Comment (ethn) [reply card/lib/card/env/location.rb:15]: I think Card.config will probably be the more approachable convention when already inside card context.

File: card-mod-delayed_job/config/initializers/delayed_job_config.rb [hunk 16]
```diff
@@ -1,6 +1,6 @@
-return unless Cardio.config.active_job.queue_adapter == :delayed_job
+return unless Card.config.active_job.queue_adapter == :delayed_job
```
Comment (ethn) [inline card-mod-delayed_job/config/initializers/delayed_job_config.rb:1]: here it has to be Cardio.  Card is not loaded yet.

Comment (ethn) [reply card-mod-delayed_job/config/initializers/delayed_job_config.rb:1]: oh, maybe it is for now. Sorry.  I want this to be loaded earlier but that's not an option yet...

File: card-mod-monkey/lib/card/pry.rb [hunk 17]
```diff
@@ -1,4 +1,4 @@
-module Cardio
+class Card
```
Comment (ethn) [inline card-mod-monkey/lib/card/pry.rb:1]: This is wrong.  Card is not loaded yet.

File: card/lib/card/config/environments/cucumber.rb [hunk 18]
```diff
@@ -0,0 +1,4 @@
+# -*- encoding : utf-8 -*-
+
+Cardio.application.class.configure do
```
Comment (ethn) [inline card/lib/card/config/environments/cucumber.rb:3]: There is no whitelist of environments.  We must get rid of these files.

File: card/lib/card/config/initializers/secret_token.rb [hunk 19]
```diff
@@ -0,0 +1,16 @@
...
+# no regular words or you'll be exposed to dictionary attacks.
+
+Rails.application.configure do
+  config.secret_token = "65632e048d6c80d1e63c911e1a40a51072413543f3182e0261"\
+                       "b52e3812b2c9f0ee81828fa5688a34c13a78e438a4b56f4971"\
+                       "35dc079a4e4460733d555968a2f8"
+  config.secret_key_base = "65632e048d6c80d1e63c911e1a40a51072413543f3182e0"\
+                           "261b52e3812b2c9f0ee81828fa5688a34c13a78e438a4b5"\
+                           "6f497135dc079a4e4460733d555968a2f8"
+end
```
Comment (ethn) [inline card/lib/card/config/initializers/secret_token.rb:16]: this can't be here!!

**Pull request closed without merging**